dd-sdk-android/src/main/kotlin/com/datadog/android/sessionreplay/internal/NoOpLifecycleCallback.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import com.datadog.android.sessionreplay.LifecycleCallback
+
+internal class NoOpLifecycleCallback : LifecycleCallback {
```

Comment:
Why don't you use the automatic NoOpFactory?

Reply:
Mostly because the `LifecycleCallback` lives in the `dd-dk-android-session-replay` module and `NoOpLifecycleCallback` is only relevant in the `dd-sdk-android` module where is actually being used. If I used the `NoOpFactory` it will be generated in the `dd-sdk-android-session-replay` and I want to avoid that.